### PR TITLE
Fix: negative BYYEARDAY indexes into the previous year

### PR DESCRIPTION
**Pull request by sunesimonsen**
Accoding to rfc5545 negative BYYEARDAY should be an offset from the end
of the current year:

https://tools.ietf.org/html/rfc5545#section-3.3.10

The BYYEARDAY rule part specifies a COMMA-separated list of days
of the year.  Valid values are 1 to 366 or -366 to -1.  For
example, -1 represents the last day of the year (December 31st)
and -306 represents the 306th to the last day of the year (March
1st).  The BYYEARDAY rule part MUST NOT be specified when the FREQ
rule part is set to DAILY, WEEKLY, or MONTHLY.

This commit fixes that problem with the minimal possible change.

### CODE REVIEW

File: test/recur_iterator_test.js [hunk 1]
```diff
@@ -879,4 +879,47 @@ suite('recur_iterator', function() {
...
     ]
   });
+
+  //yearly, byYearDay
+  testRRULE('FREQ=YEARLY;BYYEARDAY=1,2,4,6,11,12,20,42,48,49,-306,-303,' +
+            '-293,-292,-266,-259,-258,-239,-228,-209,-168,-164,-134,-133,' +
+            '-113,-105,-87,-56,-44,-26,-21,-14', {
+    dtStart: '2015-01-01',
+    dtEnd: '2015-01-02',
+    max: 32,
```
Comment (kewisch) [inline test/recur_iterator_test.js]: `dtEnd` is not handled (nor needed) and `max` is defaults to the length of dates if not specified. Can you remove the lines on `dtEnd` and `max`?

**Comment by kewisch** [comment]
Very nice, I seem to recall ical.js was differing from libical with some byyearday rules and this seems to fix it. Aside from the comment above, could you add another testcase that uses a leap year and dates from negative BYYEARDAY both before and after Feb 29th? It doesn't have as many dates, just a few should be fine.

**Comment by sunesimonsen** [comment]
Sure, will do that tomorrow :-)

**Comment by sunesimonsen** [comment]
Hi @kewisch, I found a bug when I made some more tests, so I fixed the bug and refactored to code a bit to introduce a `_nextByYearDay` method that would be used from two different places. Sorry that you have to review the code again. 

Kind regards
Sune

**Comment by sunesimonsen** [comment]
The build seems to be failing because of a travis error: 

```
1.93s$ npm install -g grunt-cli
/home/travis/.nvm/v0.10.38/bin/grunt -> /home/travis/.nvm/v0.10.38/lib/node_modules/grunt-cli/bin/grunt
grunt-cli@0.1.13 /home/travis/.nvm/v0.10.38/lib/node_modules/grunt-cli
├── resolve@0.3.1
├── nopt@1.0.10 (abbrev@1.0.6)
└── findup-sync@0.1.3 (lodash@2.4.2, glob@3.2.11)
before_install.2
0.01s$ echo -e "Host *\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
before_install.3
0.01s$ eval `ssh-agent -s`
Agent pid 2482
$ echo "$GITHUB_SSH_KEY" | base64 -d | ssh-add -
Enter passphrase for (stdin):

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
```

**Comment by kewisch** [comment]
Thanks for the note, it is indeed fallout from the recent doc changes. I'll fix this on master today.

**Comment by sunesimonsen** [comment]
Ping me when you do, then I'll rebase an force push.

**Comment by kewisch** [comment]
I've merged the fix to master now, please retry :)

**Comment by sunesimonsen** [comment]
Thanks that fixed it :-)

**Comment by kewisch** [comment]
Thanks for the patch :-)

**Comment by sunesimonsen** [comment]
Thanks for merging.

**Pull request merged**